### ISSUE TITLE
[BUG] auth0 broken due to mismatch between documentantion and code boilerplates

### ISSUE THREAD

**Issue by avallete** [hasura/learn-graphql#436]
# TL;DR
Following the backend and frontend tutorials for `react-apollo-hooks` it's impossible to connect using Auth0.

# Problem description

There is a mismatch between the server documentation and the frontend boilerplate code.

Introduced into #310 the server tutorial lead the user, on the Auth0 plateform to save the "hasura claims" into the "accessToken" object via the defined rule:

```javascript
function (user, context, callback) {
  const namespace = "https://hasura.io/jwt/claims";
  context.accessToken[namespace] =
    {
      'x-hasura-default-role': 'user',
      // do some custom logic to decide allowed roles
      'x-hasura-allowed-roles': ['user'],
      'x-hasura-user-id': user.user_id
    };
  callback(null, user, context);
}
```

However, into the frontend boilerplate code you can find this kind of calls: https://github.com/hasura/learn-graphql/blob/2c312fdf1455a3d4add908b83e644d085ec4589e/tutorials/frontend/react-apollo-hooks/app-final/src/components/Auth/react-auth0-spa.js#L41

Which will return an empty token (since `idToken` isn't set into auth0) and therefore, cause your app to fail his linking between hasura and auth0 after login/signup.

# Possibles fix:

Either:

- Rollback the change, on server side documentation, and ask the user to set the `idToken` instead of `accessToken`. Don't know if it's a good idea but reading the comment: https://github.com/hasura/learn-graphql/issues/310#issuecomment-669739958 I guess there was a reason for this change. But reading [the auth0 docs](https://auth0.com/docs/tokens/id-tokens) about `idToken` I don't see why it's not suited for this usage. Some insight from @praveenweb may be useful.
- Update the provided boilerplate frontend code, in that case, there is probably not only the  `react-apollo-hooks` which is impacted.

**Comment by praveenweb** [comment]
@avallete - Thanks for raising this issue. This issue has been open for a while. We have seen multiple updates from Auth0 SDK and its usage on the frontend that kept us from updating the usage.

Auth0's SDK (on their older versions) exposed the JWT claim via idToken and hence the initial boilerplate app was created using the code you mentioned above. Recent library updates have ensured that accessToken also generates the valid JWT claim on the frontend. The code has been fixed on the [final-app](https://github.com/hasura/learn-graphql/blob/master/tutorials/frontend/react-apollo-hooks/app-final/src/components/App.js#L33).

Closing this issue now :)
